<launch>

<arg name="map_size_x" default="50.0"/>
<arg name="map_size_y" default="50.0"/>
<arg name="map_size_z" default=" 8.0"/>

<arg name="start_x" default=" -25.0"/>
<arg name="start_y" default=" -25.0"/>
<arg name="start_z" default=" 0.0"/>

<arg name="max_vel" default="3.0"/>
<arg name="max_acc" default="2.5"/>
<arg name="resolution" value="0.5" />

<arg name="run_kino_a_star" default="true"/>

  <node pkg="kino_planner" type="kino_plan_node" name="kino_plan_node" output="screen" required = "true">
      <remap from="~waypoints"       to="/waypoint_generator/waypoints"/>
      <remap from="~map"             to="/random_forest/global_map"/> 

      <param name="occ_map/origin_x" value="$(arg start_x)"/>
      <param name="occ_map/origin_y" value="$(arg start_y)"/>
      <param name="occ_map/origin_z" value="$(arg start_z)"/>

      <param name="occ_map/map_size_x" value="$(arg map_size_x)"/>
      <param name="occ_map/map_size_y" value="$(arg map_size_y)"/>
      <param name="occ_map/map_size_z" value="$(arg map_size_z)"/>

      <param name="occ_map/resolution" value="$(arg resolution)" type="double"/>

      <param name="search/run_kino_a_star" value="$(arg run_kino_a_star)"/>
      
      <!-- kinodynamic path searching -->
      <param name="search/max_tau" value="0.5" type="double"/>
      <param name="search/init_max_tau" value="1.0" type="double"/>
      <param name="search/max_vel" value="$(arg max_vel)" type="double"/>
      <param name="search/vel_margin" value="0.25" type="double"/>
      <param name="search/max_acc" value="$(arg max_acc)" type="double"/>
      <param name="search/w_time" value="10.0" type="double"/>
      <param name="search/horizon" value="10000.0" type="double"/>
      <param name="search/lambda_heu" value="10.0" type="double"/>
      <param name="search/resolution_astar" value="0.025" type="double"/>
      <param name="search/time_resolution" value="0.8" type="double"/>
      <param name="search/margin" value="0.2" type="double"/>
      <param name="search/allocate_num" value="100000" type="int"/>
      <param name="search/check_num" value="10" type="int"/>
  </node>
</launch>